File: backend/java/localpos-user-service/src/main/java/com/pds/localpos/userservice/repository/StoreRepository.java
package com.pds.localpos.userservice.repository;

import com.pds.localpos.userservice.model.Store;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface StoreRepository extends JpaRepository<Store, String> {

    Set<Store> findByCodeIn(Set<String> codes);

    Optional<Store> findByCode(String code);
}

File: backend/kotlin/localpos-user-service-kt/src/main/kotlin/com/pds/localpos/userservice/repository/StoreRepository.kt
package com.pds.localpos.userservice.repository

import com.pds.localpos.userservice.model.Store
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface StoreRepository : JpaRepository<Store, String> {

    fun findByCodeIn(codes: Collection<String>): Set<Store>
}
